react-hash-routing/client/app.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import Header from './components/header';
+import Catalog from './pages/catalog';
+import NotFound from './pages/not-found';
+import ProductDetails from './pages/product-details';
+import { parseRoute } from './lib';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      route: parseRoute(window.location.hash)
+    };
+  }
+
+  componentDidMount() {
+    /**
+     * Listen for hash change events on the window object
+     * Each time the window.location.hash changes, parse
+     * it with the parseRoute() function and update state
+     */
+    const initialLocationHash = window.location.hash;
+    window.addEventListener('hashchange', () => {
+      if (initialLocationHash !== window.location.hash || !initialLocationHash) {
```

Comment:
You can remove this conditional check. The hashchange event only fires when the hash changes 😂